vortex-expr/src/select.rs

```diff
@@ -0,0 +1,85 @@
+use std::any::Any;
+use std::collections::HashSet;
+
+use vortex::array::StructArray;
+use vortex::variants::StructArrayTrait;
+use vortex::{Array, IntoArray};
+use vortex_dtype::field::Field;
+use vortex_error::{vortex_bail, VortexResult};
+
+use crate::{unbox_any, Column, VortexExpr};
+
+// TODO(robert): Support wildcard and column exclusions
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Select {
```

Comment:
nit but i thought we call this sort of thing `Project` in a few other places

Reply:
We do but there's a good argument that Project is an overloaded sql name. Also now it's a bit more explicit that it's different